manifests/instance.pp

```diff
@@ -73,6 +78,7 @@
   $status             = $elasticsearch::status,
   $config             = undef,
   $configdir          = undef,
+  $manage_datadir     = true,
```

Comment:
Are you sure?  If I have it default to $elasticsearch::manage_datadir, I will still have the problem that the user may override the default either in the elasticsearch class, or in the elasticsearch::instance type.  Of course, I'd rather just force to user to set it in the elasticsearch::instance type, however it appears to me from the design that the preference is to allow the user to set things in two places.